pkg/pillar/kubeapi/nodedrain.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package kubeapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+)
+
+// RequestNodeDrain generates the NodeDrainRequest object and publishes it
+func RequestNodeDrain(pubNodeDrainRequest pubsub.Publication, requester DrainRequester, context string) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("RequestNodeDrain: can't get hostname %v", err)
+	}
+	drainReq := NodeDrainRequest{
+		Hostname:    hostname,
+		RequestedAt: time.Now(),
+		RequestedBy: requester,
+		Context:     context,
+	}
+	err = pubNodeDrainRequest.Publish("global", drainReq)
+	if err != nil {
+		return fmt.Errorf("RequestNodeDrain: error publishing drain request: %v", err)
+	}
+	return nil
+}
+
+// GetDrainStatusOverride : an alternate way to set drain status for debug
+func GetDrainStatusOverride() *NodeDrainStatus {
+	// An alternate path to force a drain status in the event of a drain issue.
+	forceNodeDrainPath := "/persist/force-NodeDrainStatus-global.json"
+	if _, err := os.Stat(forceNodeDrainPath); err == nil {
+		b, err := os.ReadFile(forceNodeDrainPath)
+		if err == nil {
+			cfg := NodeDrainStatus{}
+			err = json.Unmarshal(b, &cfg)
+			if err == nil {
+				if cfg.Status == COMPLETE {
+					os.Remove(forceNodeDrainPath)
```

Comment:
I heard you like indentation ;-)

```suggestion
					err := os.Remove(forceNodeDrainPath)
					if err != nil {
						log.Warnf("could not remove %s: %v", forceNodeDrainPath, err)
					}
```

Reply:
Yeah this could be shortened a bit, just a moment.